.azure-pipelines/templates/steps/tox-steps.yml

```diff
@@ -0,0 +1,46 @@
+steps:
+  - bash: |
+      brew install augeas
+    condition: startswith(variables['IMAGE_NAME'], 'macOS')
+    displayName: Install MacOS dependencies
+  - bash: |
+      sudo apt-get update
+      sudo apt-get install -y --no-install-recommends \
+        python-dev \
+        gcc \
+        libaugeas0 \
+        libssl-dev \
+        libffi-dev \
+        ca-certificates \
+        nginx-light \
+        openssl \
+        expect
+      sudo systemctl stop nginx
+    condition: startswith(variables['IMAGE_NAME'], 'ubuntu')
+    displayName: Install Linux dependencies
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: $(PYTHON_VERSION)
+      addToPath: true
+    condition: ne(coalesce(variables['PYTHON_VERSION'], 'UNSPECIFIED'), 'UNSPECIFIED')
```

Comment:
Yes `coalesce` is a typical function in the SQL world, way less in imperative languages since `if/else` + boolean logic.

Surprinsingly, `not(variables['PYTHON_VERSION'])`, despite `variables['PYTHON_VERSION']` being evaluated as the `Null` object, and the `Null` object stated to be casted as the `Boolean` `false` value, for the purpose of the `not()` function.

But the conversion of `Null` into the `String` `''` is working, so `ne(variables['PYTHON_VERSION'], '')` is working as expected, so I use that.